airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskScopeProvider.kt

```diff
@@ -57,42 +51,49 @@ class DestinationTaskScopeProvider(config: DestinationConfiguration) :
     private val timeoutMs = config.gracefulCancellationTimeoutMs
 
     data class ControlScope(
-        val job: Job,
+        val name: String,
+        val job: CompletableJob,
         val dispatcher: CoroutineDispatcher,
-        val scope: CoroutineScope = CoroutineScope(dispatcher + job)
+        val scope: CoroutineScope = CoroutineScope(dispatcher + job),
+        val runningJobs: AtomicLong = AtomicLong(0)
     )
 
-    private val internalScope = ControlScope(Job(), Dispatchers.IO)
+    private val internalScope = ControlScope("internal", Job(), Dispatchers.IO)
 
     private val implementorScope =
         ControlScope(
-            SupervisorJob(),
+            "implementor",
+            Job(),
             Executors.newFixedThreadPool(config.maxNumImplementorTaskThreads)
                 .asCoroutineDispatcher()
         )
 
     override suspend fun launch(task: WrappedTask<ScopedTask>) {
         when (task.innerTask) {
-            is InternalScope -> internalScope.scope.launch { execute(task, "internal") }
-            is ImplementorScope -> implementorScope.scope.launch { execute(task, "implementor") }
-            is ShutdownScope -> implementorScope.scope.launch { execute(task, "shutdown") }
+            is InternalScope -> internalScope.scope.launch { execute(task, internalScope) }
+            is ImplementorScope -> implementorScope.scope.launch { execute(task, implementorScope) }
         }
     }
 
-    private suspend fun execute(task: WrappedTask<ScopedTask>, scope: String) {
-        log.info { "Launching task $task in scope $scope" }
+    private suspend fun execute(task: WrappedTask<ScopedTask>, scope: ControlScope) {
+        var nJobs = scope.runningJobs.incrementAndGet()
+        log.info { "Launching task $task in scope ${scope.name} ($nJobs now running)" }
         val elapsed = measureTimeMillis { task.execute() }
-        log.info { "Task $task completed in $elapsed ms" }
+        nJobs = scope.runningJobs.decrementAndGet()
+        log.info { "Task $task completed in $elapsed ms ($nJobs now running)" }
     }
 
-    override suspend fun close() {
-        log.info { "Closing task scopes" }
+    override suspend fun close(): Unit = coroutineScope {
```

Comment:
for my understanding: why does this need to be in a new scope?

Reply:
It doesn't need to be. I added that when I was debugging because I tried launching the join/cancel in a `launch` block, which requires a scope. (This method creates a new scope object that is a child of the calling scope and inherits its context.)

It's no longer necessary, removed.